### PR TITLE
Component sample README files

### PR DESCRIPTION
**Pull request by DukeDavis12**
Component sample README files.

Signed-off-by: Davis Benny <davis.benny@intel.com>

### CODE REVIEW

File: component-samples/demo/README.MD [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+```
+sudo docker system prune -a
+```
+
+# Podman Commands
+
+## Start Podman
+* Use the following command to start the podman container.
+```
+podman-compose up -d --build
```
Comment (shrikant1407) [inline component-samples/demo/README.MD:81]: sudo will required with podman also

File: component-samples/demo/README.MD [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+```
+
+## Stop Podman
+
+* Use the following command to stop a specific podman container.
+```
+podman stop <container-name>
```
Comment (shrikant1407) [inline component-samples/demo/README.MD:88]: Add sudo

File: component-samples/demo/README.MD [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+```
+OR
+```
+podman stop <container-id>
```
Comment (shrikant1407) [inline component-samples/demo/README.MD:92]: Add sudo

File: component-samples/demo/README.MD [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+```
+podman rmi <image-id>
+```
+
+* Use the following command to delete all the podman artifacts. (**Note:** podman containers must be stopped before deleting them)
+```
+podman system prune -a
+```
+# Configuring Proxies
+Update the proxy information in `_JAVA_OPTIONS` as
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:125]: Should we consolidate all proxy information in one place?
Also, lets talk about this readme and its path

File: component-samples/demo/README.MD [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+are printed in the log file.
+
+Use following command to extract the log file from a running container.
+```
+docker container cp <container-id>:/home/fdo/log-filename .
+```
+
+# Running Demo
+
+***NOTE***: Use the following commands to enbale FDO support on RHEL.
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:177]: Enable - typo

File: component-samples/demo/README.MD [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+```
+bash scripts/enable_rhel_support.sh
+echo $'\nexport PODMAN_USERNS=keep-id' >> ~/.bashrc
+source ~/.bashrc
+```
+
+1. Start the FDO Manufacturer Sample as per the steps outlined in [Manufacturer README](manufacturer/README.md).
+
+2. Complete Device Initialization (DI) by starting the FDO HTTP Java Device Sample as per the steps outlined in [Device README](device/README.md). Delete any existing 'credentials.bin' before starting the device.
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:186]: Is it credentials.bin or credential,bin?

File: component-samples/demo/README.MD [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+
+3. Complete Ownership Voucher Extension by using the API `GET /api/v1/mfg/vouchers/<serial_no>` and save the Ownership Voucher. Please refer to [PRI Manufacturer REST APIs](manufacturer/README.md/#fdo-pri-manufacturer-rest-apis) for more information about the API.
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:188]: Should we also list the commands?

File: component-samples/demo/README.MD [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+6. Complete Transfer Ownership 1 and 2 (TO1 and TO2) by starting the FDO PRI HTTP Java Device Sample again. The previously created 'credentials.bin' from Step#2 will be used directly by the Device.
+
+***NOTE***: Credential Reuse protocol is enabled by default, that is, after a successful onboarding the device credentials remain unchanged. To enable the Resale protocol instead, wherein, after a successful onboarding the device credentials are replaced, update the fields `REPLACEMENT_GUID` and/or `REPLACEMENT_RVINFO` in Owner `TO2_DEVICES` table by using the API `POST /api/v1/owner/setupinfo?id=<device_guid>` before starting TO2. Please refer to [FDO PRI Owner REST APIs](owner/README.md/#fdo-pri-owner-rest-apis) for more information about the API.
+
+# ServiceInfo setup between FDO PRI HTTP Java Device Sample and FDO PRI Owner Sample
+
+The FDO PRI HTTP Java Device Sample currently supports `fdo_sys` and `fidoalliance` module for interpreting received owner ServiceInfo and `devmod` module to share device ServiceInfo with Owner.
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:200]: Should we call out fido_alliance support?

File: component-samples/demo/README.MD [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+
+- `fdo_sys` Owner ServiceInfo module: This module supports the following 3 message names as listed below to interpret the ServiceInfo as received from the Owner. The basic functionality of this module is to support payload/script transfers and basic command execution.  A sample format looks like 'fdo_sys:filedesc=filename, fdo_sys:write=filecontent,fdo_sys:exec=command-to-execute'.
+
+    *filedesc* - The name to be given to the file once it is transferred. Upon receiving this, device creates a file with the given name and opens stream to write into it.
+
+    *write* - The payload/content (script, binaries, and others) that is sent to the device. Upon receiving this, device writes the content into the open stream as given by the preceding 'filedesc' message.
+
+    *exec* - The command that will be executed at the device. Device executes the command as received.
+
+    *exec-cb* - This command returns the status / progress of the process running.
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:210]: Do we need to give any examples?

File: component-samples/demo/README.MD [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+
+- `devmod` Device ServiceInfo module: This module supports multiple messages as listed down in the protocol specification, that is sent to the Owner as Device ServiceInfo. A sample format looks like 'devmod:active=1'.
+
+- Ensure that before each module the `module_name:active` message is sent as `true` for a successful ServiceInfo transfer. The device will ignore all ServiceInfo module messages if module_name:active doesn't precede the messages of the module.
+
+The FDO PRI Owner Sample currently supports the same `fdo_sys` module to send Owner ServiceInfo to the Device and `devmod` module to store the received Device ServiceInfo. Few sample ServiceInfo values, as present in owner/serviceinfo/sample-values/ are populated into the database table 'OWNER_SERVICEINFO' as byte arrays. For keeping the association between the Device and the ServiceInfo values to transfer, 'GUID_OWNERSVI' database table is used. When a Device is inserted into the database table 'TO2_DEVICES', it'll not have any association with the ServiceInfo values, and so by default, no ServiceInfo is transferred to the Device.
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:219]: Should we call out how database can be accessed?

File: component-samples/demo/README.MD [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+To enable ServiceInfo transfer to a Device with a given GUID, follow the steps below:
+
+Insert required ServiceInfo resources into the database table 'SYSTEM_MODULE_RESOURCE' using the API `POST /api/v1/owner/resource?filename=fileName`. More information about the same is provided in section [FDO PRI Owner REST APIs](owner/README.md/#fdo-pri-owner-rest-apis). If the required ServiceInfo already exists in the table with appropriate tags, start TO1.
+
+
+# Customize for multi-machine setup
+
+While testing for a scenario where different components are running in different
+machines, the SSL certificate needs to be updated to include the DNS / IP of
+these machines in its SubjectAltName.
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:232]: Should we call out where they need to make this change?

File: component-samples/demo/aio/README.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
+# About
+
+The FDO All-In-One demo is designed to run multiple FDO services like Manufacturer, Rendezvous and Owner Service in a single
+**docker** instance. The purpose of this demo is to fast-track demonstration of FDO protocol using different client implementations.
+
+***NOTE***: The All-In-One demo is provided to demonstrate out-of-box operation of FDO components.  Appropriate security measures with respect to key-store management and credential management should be considered while performing production deployment of any FDO component.
+
+# Getting Started with the FDO All-In-One Demo
+
+The following are the system constraints for the All-in-One demo.
```
Comment (v-vasavi) [inline component-samples/demo/aio/README.md:10]: Are these system constraints or system requirements?

File: component-samples/demo/aio/README.md [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+
+All the runtime configurations for the services are specified in four files: `service.env`, `hibernate.cfg.xml`, `service.yml` and `WEB-INF/web.xml`.
+
+`service.env`: consists of all the credentials used by the All-in-one demo service. These credential configurations are to be generated freshly for each deployment.
+
+`hibernate.cfg.xml`: consists of all the database configurations used by the All-in-one demo service. This file can be configured to pick various database tables and properties.
+
+`service.yml` file is structured into multiple sections:
+
+- `hibernate-properties:` - This section contains hibernate related runtime properties including the DB URL, dialect and others.
```
Comment (v-vasavi) [inline component-samples/demo/aio/README.md:46]: Might be good to capitalize "H" in Hibernate in all places

File: component-samples/demo/aio/README.md [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+| GET /api/v1/owner/svisize | Collects the owner svi size from `ONBOARDING_CONFIG` table | | text/plain |  | MAX_MESSAGE_SIZE |
+| POST /api/v1/owner/svisize | Updates the owner svi size in `ONBOARDING_CONFIG` table | | text/plain | MAX_MESSAGE_SIZE | |
+| GET /api/v1/owner/resource?filename=fileName | Returns the file based on filename from `SYSTEM_RESOURCE` table | Path - filename | | | file |
+| POST /api/v1/owner/resource?filename=fileName | Adds the file to DB based on filename  from `SYSTEM_RESOURCE` table  | Path - filename | text/plain| file in Binary format |  |
+| DELETE /api/v1/owner/resource?filename=fileName | Delete the  file from DB based on filename from `SYSTEM_RESOURCE` table   | Path - filename | | |  |
+
+
+***NOTE***: These REST APIs use Digest authentication. `api_user` and `api_password` properties specify the credentials to be used while making the REST calls.
+
+Following is the list of REST response error codes and it's description :
```
Comment (v-vasavi) [inline component-samples/demo/aio/README.md:159]: and their "possible" reasons?

File: component-samples/demo/aio/README.md [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+| GET /api/v1/owner/messagesize | Collects the max message size from `ONBOARDING_CONFIG` table | | |  | MAX_MESSAGE_SIZE |
+| POST /api/v1/owner/messagesize | Updates the max message size in `ONBOARDING_CONFIG` table | | | MAX_MESSAGE_SIZE | |
+| GET /api/v1/owner/svisize | Collects the owner svi size from `ONBOARDING_CONFIG` table | | text/plain |  | MAX_MESSAGE_SIZE |
+| POST /api/v1/owner/svisize | Updates the owner svi size in `ONBOARDING_CONFIG` table | | text/plain | MAX_MESSAGE_SIZE | |
+| GET /api/v1/owner/resource?filename=fileName | Returns the file based on filename from `SYSTEM_RESOURCE` table | Path - filename | | | file |
+| POST /api/v1/owner/resource?filename=fileName | Adds the file to DB based on filename  from `SYSTEM_RESOURCE` table  | Path - filename | text/plain| file in Binary format |  |
+| DELETE /api/v1/owner/resource?filename=fileName | Delete the  file from DB based on filename from `SYSTEM_RESOURCE` table   | Path - filename | | |  |
+
+
+***NOTE***: These REST APIs use Digest authentication. `api_user` and `api_password` properties specify the credentials to be used while making the REST calls.
```
Comment (v-vasavi) [inline component-samples/demo/aio/README.md:157]: Do we need to have a pointer to how / where they can find / set this?

File: component-samples/demo/manufacturer/README.md [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+
+Appropriate proxy configuration should be updated in **`_JAVA_OPTIONS`** environment variable. (Mandatory, if you are working behind a proxy.)
+
+Update the proxy information in _JAVA_OPTIONS as ```_JAVA_OPTIONS=-Dhttp.proxyHost=http_proxy_host -Dhttp.proxyPort=http_proxy_port -Dhttps.proxyHost=https_proxy_host -Dhttps.proxyPort=https_proxy_port```.
+
+# Getting the Executable
+
+Use the following commands to build FIDO Device Onboard (FDO) source.
+For the instructions in this document, `<fdo-pri-src>` refers to the path of the FDO PRI folder 'pri-fidoiot'.
+```
```
Comment (v-vasavi) [inline component-samples/demo/manufacturer/README.md:28]: Lets chat on this step details

File: component-samples/demo/README.MD (hunk 6 above)
Comment (DukeDavis12) [reply component-samples/demo/README.MD:186]: credentials.bin

File: component-samples/demo/README.MD (hunk 7 above)
Comment (DukeDavis12) [reply component-samples/demo/README.MD:188]: Actually, we have a script to download extended voucher and to upload voucher to Owner.

File: component-samples/demo/README.MD (hunk 9 above)
Comment (DukeDavis12) [reply component-samples/demo/README.MD:210]: We can add a sample SVI instruction using all commands

File: component-samples/demo/README.MD [hunk 17]
```diff
@@ -0,0 +1,233 @@
...
+Use following command to extract the log file from a running container.
+```
+docker container cp <container-id>:/home/fdo/log-filename .
+```
+
+# Running Demo
+
+***NOTE***: Use the following commands to enbale FDO support on RHEL.
+```
+bash scripts/enable_rhel_support.sh
```
Comment (shrikant1407) [inline component-samples/demo/README.MD:179]: bash scripts/enable_podman_support.sh

File: component-samples/demo/README.MD [hunk 18]
```diff
@@ -209,25 +230,88 @@ The FDO PRI HTTP Java Device Sample currently supports `fdo_sys` and `fidoallian
 
     *exec-cb* - This command returns the status / progress of the process running.
 
+    Sample SVI instruction :
+  
+    `{"filedesc" : "sample.txt","resource" : "https://www.google.com"},
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:235]: Not sure if using google.com requires any disclaimer . can we just say "website address"

File: component-samples/demo/README.MD [hunk 19]
```diff
@@ -209,25 +230,88 @@ The FDO PRI HTTP Java Device Sample currently supports `fdo_sys` and `fidoallian
...
 
-The FDO PRI Owner Sample currently supports the same `fdo_sys` module to send Owner ServiceInfo to the Device and `devmod` module to store the received Device ServiceInfo. Few sample ServiceInfo values, as present in owner/serviceinfo/sample-values/ are populated into the database table 'OWNER_SERVICEINFO' as byte arrays. For keeping the association between the Device and the ServiceInfo values to transfer, 'GUID_OWNERSVI' database table is used. When a Device is inserted into the database table 'TO2_DEVICES', it'll not have any association with the ServiceInfo values, and so by default, no ServiceInfo is transferred to the Device.
+The FDO PRI Owner Sample currently supports the same `fdo_sys` module to send Owner ServiceInfo to the Device and `devmod` module to store the received Device ServiceInfo. When a Device is inserted into the database table 'ONBOARDING_CONFIG', it'll not have any association with the ServiceInfo values, and so by default, no ServiceInfo is transferred to the Device.
```
Comment (v-vasavi) [inline component-samples/demo/README.MD:245]: what does "Same" fdo_sys signify?

File: component-samples/demo/owner/README.md [hunk 20]
```diff
@@ -43,7 +43,7 @@ All the runtime configurations for the services are specified in four files: `se
 
 `service.yml` file is structured into multiple sections:
 
-- `hibernate-properties:` - This section contains hibernate related runtime properties including the DB URL, dialect and others.
+- `hibernate-properties:` - This section contains *Hibernate related runtime properties including the DB URL, dialect and others.
```
Comment (v-vasavi) [inline component-samples/demo/owner/README.md:46]: Should it be *Hibernate or Hibernate*

File: component-samples/demo/manufacturer/README.md [hunk 21]
```diff
@@ -89,36 +89,34 @@ In case you need super user access, prefix 'sudo -E' to above command.
...
+| GET /api/v1/certificate/validity | Collects certificate validity days from  `CERTIFICATE_VALIDITY` table | |  | | Number of Days| curl  -D - --digest -u ${api_user}: --location --request GET 'http://localhost:8039/api/v1/certificate/validity' --header 'Content-Type: text/plain' |
+| GET /health | Returns the health status |  |  | | Current version |  curl  -D - --digest -u ${api_user}:  --location --request GET 'http://localhost:8039/health' --header 'Content-Type: text/plain' |
+
+***NOTE***: These REST APIs use Digest authentication. `api_user` and `api_password` properties specify the credentials to be used while making the REST calls. The value for `api_user` is present in `service.yml` file and value for `api_password` is present in `service.env` file.
+
+Following is the list of REST response error codes and it's possible causes :
+
+|     Error Code     |             causes                  |
```
Comment (v-vasavi) [inline component-samples/demo/manufacturer/README.md:112]: Can the heading also be Possible causes?

File: component-samples/demo/README.MD (hunk 18 above)
Comment (DukeDavis12) [reply component-samples/demo/README.MD:235]: Updated

File: component-samples/demo/README.MD (hunk 19 above)
Comment (DukeDavis12) [reply component-samples/demo/README.MD:245]: Typo. Updated:
The FDO PRI Owner Sample currently supports `fdo_sys` module for sending ServiceInfo to the PRI Device and `devmod` module to store the received Device ServiceInfo. When a Device is inserted into the database table 'ONBOARDING_CONFIG', it'll not have any association with the ServiceInfo values, and so by default, no ServiceInfo is transferred to the Device.

File: component-samples/demo/owner/README.md (hunk 20 above)
Comment (DukeDavis12) [reply component-samples/demo/owner/README.md:46]: *Hibernate -- For external dependencies, we add asterisk `*`

**Review by himanshu7196 — APPROVED** [review]

**Review by bprashan — APPROVED** [review]

**Pull request merged**